### DISCUSSION TITLE
How do I add click and drag between columns on a table to reorder columns

### DISCUSSION THREAD

**Discussion by bu3alwa** [adobe/react-spectrum#6191] [Q&A]
I made a sandbox and issue is when you add the drag button in the column it does not trigger on the column. Is this a bug? I understand this feature that is not provided but I would think I would be able to extend columns to implement my own features.

https://codesandbox.io/p/devbox/laughing-mirzakhani-25fyt7

**Comment by snowystinger** [comment]
I think you need to make the codesandbox public?

**Reply by bu3alwa** [reply]
Sorry wasn't aware. They had a share button and a link to copy. Didn't think share wouldn't be sharable lol. Is it working now?

**Reply by snowystinger** [reply]
quick scan, looks like your ref is being applied to both the drag and drop elements?

that said, we haven't tried building this, so I have no idea if you'll run into other issues. we support a menu trigger in the column header, but haven't tried something like this

**Reply by bu3alwa** [reply]
I updated to fix it. It seems like I get the same behavior that I get in my app if you do it through keyboard you can get the button to become draggable but haven't been able to drop it. But on the mouse it doesn't seem to get into the drag state.

**Reply by snowystinger** [reply]
It looks like this `usePress` is the culprit https://github.com/adobe/react-spectrum/blob/main/packages/%40react-aria/table/src/useTableColumnHeader.ts#L52
It calls some preventDefaults/stopPropagations which are likely interferring.

I have a vague memory of not applying the returned value from usePress when it should be disabled, but, that caused a different problem. Someone will have to try that out and see if it causes any issues.

Either way, I think this can be regarded as a bug.

**Reply by snowystinger** [reply]
I've opened an issue https://github.com/adobe/react-spectrum/issues/6195

**Reply by bu3alwa** [reply]
Thanks Ill track that issue

**Reply by bri-fox** [reply]
Did you get this working @bu3alwa

**Reply by bu3alwa** [reply]
@bri-fox seems its a bug so track #6195
